lib/Rozier/app/custom-elements/RzSearch.ts

```diff
@@ -0,0 +1,253 @@
+import type { RzDialog } from './RzDialog'
+import type { NodeSourceSearch } from '~/types/node-source-search'
+import api from '~/api'
+import { rzCardRenderer } from '~/utils/component-renderer/rzCard'
+import { debounce } from 'lodash'
+
+const SEARCH_QUERY = 'search_all'
+
+/* Used for component documentation */
+/*
+type RzSearchAttributes = {
+    'initial-value'?: string // Initial value to populate the search input
+    'open-key'?: string // Key combination to open the search dialog (e.g., "Meta+K")
+    'status-wrapper'?: string // Wrapper element for status message and spinner wrapper
+    'results-wrapper'?: string // UL element to display search results
+    'idle-text'?: string // Text to display when idle
+    'reset-text'?: string // Text to display when search is reset
+    'pending-text'?: string // Text to display when search is pending
+    'unique-result-text'?: string // Text to display when exactly one result is found
+    'results-text'?: string // Text to display when multiple results are found
+    'no-results-text'?: string // Text to display when no results are found
+    'error-text'?: string // Text to display when an error occurs
+}
+*/
+
+export class RzSearch extends HTMLElement {
+    value: string | null = null
+    searchInput: HTMLInputElement | null = null
+
+    dialogElement: RzDialog | null = null
+    listElement: HTMLUListElement | null = null
+
+    spinnerElement: HTMLElement | null = null
+    messageElement: HTMLElement | null = null
+    fetchStatus:
+        | 'idle'
+        | 'reset'
+        | 'pending'
+        | 'results'
+        | 'no-results'
+        | 'error' = 'idle'
+
+    items: NodeSourceSearch[] | null = null
+
+    constructor() {
+        super()
+
+        this.onInputChange = debounce(this.onInputChange.bind(this), 300)
+        this.onKeyDown = this.onKeyDown.bind(this)
+    }
+
+    updateSpinnerVisibility() {
+        if (!this.spinnerElement) return
+
+        if (this.fetchStatus === 'pending') {
+            this.spinnerElement.style.display = 'initial'
+        } else {
+            this.spinnerElement.style.display = 'none'
+        }
+    }
+
+    updateStatusMessage() {
+        if (!this.messageElement) return
+
```

Comment:
extract the message getter into a another function like `getMessageText()` then
```suggestion
const text = this.getMessageText()
```